Cython/Compiler/PyrexTypes.py

```diff
@@ -3807,8 +3807,8 @@ def create_from_py_utility_code(self, env):
                 'type': self.cname,
             })
             # Override directives that should not be inherited from user code.
-            # TODO: filter directives with an allow list to keep only those that are safe and relevant.
-            directives = dict(env.directives, cpp_locals=False)
+            from . import Options
+            directives = Options.get_conversion_utility_code_directives(env.directives)
             from .UtilityCode import CythonUtilityCode
             env.use_utility_code(CythonUtilityCode.load(
```

Comment:
Would it make sense to make the `CythonUtilityCode` directly responsible for this?

Reply:
Not sure about this. It seems wrong to me for `CythonUtilityCode` to ignore some of the directives that you've explicitly passed it. I can imagine that leading to a confusing debugging session sometime in the future.

Maybe if the `compiler_directives` keyword argument was split into `explicit_directives` (which are always used as-is) and `filterable_directives` (which pass through this process)? But that seems needlessly complicated.

Reply:
Right, that would complicate things.

But when I ask when I would use this filtering, then the sole answer seems to be: "when I want to pass directives into the `CythonUtilityCode`. So the two appear highly related and the selection very specific for this purpose. Maybe `CythonUtilityCode` could have a staticmethod `filter_directives()` to do only the filtering. Simply the length and specificness of the function name suggests to me that it would benefit from namespacing.

Reply:
Yeah that makes sense (and I've done it). I think I misunderstood what you were suggesting initially.